### PR TITLE
Feature/slop 75/I'm feeling sloppy functionality

### PR DESCRIPTION
**Pull request by fdlai**
## Describe your changes
Add a MovieContextProvider and a SoundContextProvider component. Add a click handler to the **I'm feeling sloppy** button, which routes the user to the homepage opens up a random movie's modal, and plays a hardcoded sound. 

## Issue ticket number and link
slop-75
https://tripleten-apiary.atlassian.net/jira/software/projects/SLOP/boards/2?selectedIssue=SLOP-75

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [ ] I have made any necessary updates to Storybook to accompany these changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by skovakina — APPROVED** [review]
Hi! I tested everything and, I think, it works fine. Good job!

**Review by marzahlde09 — APPROVED** [review]
I believe everything works, and I was able to implement your sound context successfully!

File: src/components/header/header.jsx [hunk 1]
```diff
@@ -30,6 +39,17 @@ export const Header = () => {
     openModal(<SignupModal onClose={closeModal} />);
   }
 
+  function openMoviePreviewModal() {
+    openModal(
```
Comment (tomershvueli) [inline src/components/header/header.jsx:43]: There's a (slight) chance that `getRandomMovie` would return us `undefined` and the `MoviePreviewModal` does rely on there being a `selectedMovie` present. Can we call `getRandomMovie` outside of the `openModal` call and only conditionally `openModal` if we get a movie in response?

File: src/components/header/header.jsx [hunk 2]
```diff
@@ -88,27 +109,39 @@ Header.NavLinks = ({
...
   openLoginModal,
+  openMoviePreviewModal,
 }) => {
+  const { playSound } = useSounds();
   const navLinks = [
     {
       title: "Slop Search",
       src: headerMagnifyglass,
       link: "/search",
+      onClick: () => {},
```
Comment (tomershvueli) [inline src/components/header/header.jsx:120]: Does adding an `onClick` to one link mean we _have_ to define it for all the others, even if we don't need them? Ideally we don't add code that we don't need/that has no purpose.

File: src/components/header/header.jsx (hunk 1 above)
Comment (fdlai) [reply src/components/header/header.jsx:43]: Sure, I've updated the code with that change!

File: src/components/header/header.jsx (hunk 2 above)
Comment (fdlai) [reply src/components/header/header.jsx:120]: No, it's not necessary to have an empty onClick function for the other links. I've updated the code to remove them!

**Review by tomershvueli — APPROVED** [review]
Thanks for addressing my earlier comments - this is looking great!

**Pull request merged**